scripts/build.sh

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+script_name="${BASH_SOURCE:-$0}"
+script_path=$(realpath "$script_name")
+scripts_dir_path=$(dirname "$script_path")
+
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_functions.sh"
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_logger.sh"
+
+function usage {
+    cat <<EOF
+This script performs the following tasks in this specific order:
+1. Generates a list of Operator manifests for a SonataFlow project
+2. Builds the workflow image using 
+3. Optionally deploys.
+    3.1 Pushes the image to a registry (specidied in the image path)
+    3.2 Applies the generated manifests in the current k8s namespace
+
+Usage: 
+    $script_name [flags]
+
+Flags:
+    -i|--image string                 The image path to use for the workflow (required).
+    -b|--builder-image string         Overrides the image to use for building the workflow image.
+    -h|--help                         Prints this help message.
+    -r|--runtime-image string         Overrides the image to use for running the workflow.
+    -w|--workflow-directory string    Path to the directory containing the workflow's files (the 'src' directory). Default: current directory.
+    -P|--no-persistence               Skips adding persistence configuration to the sonataflow CR.
+       --apply                        Applies the generated manifests in the current namespace.
```

Comment:
Suggestion: I would name this arg `deploy` instead of apply as to "deploy the application" and it would match the wording you used earlier in the description

Reply:
But then `--deploy` would have to build and apply the CRs. No freedom, no choice, it's all or nothing...

The thing is that `--apply` doesn't really deploy "everything". The function you refer to is called "maybe_deploy" because in some cases (when neither `--push` nor `--apply` are specified) it might do nothing.
The rationale was that maybe you would like only to push a new image without applying the CRs or maybe you might want to apply the CRs without building a new image (maybe you will use someone else's image, you just need the manifests to be generated with a different image).

If it feels more convenient, I can add `--deploy` that would be the equivalent to `--push` && `--apply`. WDYT?

Reply:
> If it feels more convenient, I can add --deploy that would be the equivalent to --push && --apply. WDYT?

I think it's a good idea

>  The function you refer to is called "maybe_deploy" because in some cases (when neither --push nor --apply are specified) it might do nothing

yeah, this function bothers me a little bit as it does 2 things based on 2 differents flags, so the "unitary" of the function is lost
But I won't "fight" you on this, it's a very little thing and I am OK to keep it as it is